### PR TITLE
Health check validates policy branches

### PR DESCRIPTION
**Pull request by BradleyBoutcher**
### What does this PR do?
- Add validation logic to health-check.rb for missing non-root policy branches
- Add tests for policy branch validation

### What ticket does this PR close?
Connected to #132 

#### Change log
- [X] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [X] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [X] This PR does not require updating any documentation

### CODE REVIEW

File: bin/health-check.rb [hunk 1]
```diff
@@ -44,6 +50,19 @@ def error(message)
...
 
+if ConjurClient.policy != 'root'
+  begin
+    policy_resource = policy_resource(master_api)
+
+    policy_resource_exists = policy_resource.exists
+  rescue
+    error(
+        "Error: The policy branch specified in your configuration does not exist," \
+      " or is incorrect. Please verify that your policy exists and try again."
```
Comment (izgeri) [inline bin/health-check.rb:61]: indentation is off

File: features/health-check.feature [hunk 2]
```diff
@@ -14,13 +14,26 @@ Feature: Health Check
     Then the output includes 'There is an issue with your Conjur configuration.'
     And the exit status should be 1
 
+  Scenario: Policy Branch is Set
+    When I run the health check script with env CONJUR_FOLLOWER_URL=$CONJUR_APPLIANCE_URL CONJUR_POLICY=$CONJUR_POLICY
+    Then the exit status should be 0
```
Comment (izgeri) [inline features/health-check.feature:19]: can we also check the output?

File: features/health-check.feature [hunk 3]
```diff
@@ -14,13 +14,26 @@ Feature: Health Check
     Then the output includes 'There is an issue with your Conjur configuration.'
     And the exit status should be 1
 
+  Scenario: Policy Branch is Set
+    When I run the health check script with env CONJUR_FOLLOWER_URL=$CONJUR_APPLIANCE_URL CONJUR_POLICY=$CONJUR_POLICY
```
Comment (izgeri) [inline features/health-check.feature:18]: is the follower url needed on this line? could you still be successful just setting policy?

File: features/health-check.feature [hunk 4]
```diff
@@ -14,13 +14,26 @@ Feature: Health Check
...
+    Then the exit status should be 0
+
+  Scenario: Policy Branch is Not Set
+    When I run the health check script with env CONJUR_FOLLOWER_URL=$CONJUR_APPLIANCE_URL CONJUR_POLICY=
+    Then the exit status should be 0
```
Comment (izgeri) [inline features/health-check.feature:23]: can we also check the output?

File: features/health-check.feature [hunk 5]
```diff
@@ -14,13 +14,26 @@ Feature: Health Check
...
+
+  Scenario: Policy Branch Doesn't Exist
+    When I run the health check script with env CONJUR_FOLLOWER_URL=$CONJUR_APPLIANCE_URL CONJUR_POLICY=nonexistent_policy
```
Comment (izgeri) [inline features/health-check.feature:26]: is the follower url needed on this line, or would this test case still be valid if you dropped it?

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Service broker returns 404 when the org / space policy branches do not exist
   as expected with a helpful error message, rather than returning 500.
   [cyberark/conjur-service-broker#192](https://github.com/cyberark/conjur-service-broker/issues/192)
+- Service broker validates the existence of any provided non-root policy,
+  rather than providing an ambiguous configuration error.
```
[the remark this answers is not in the document]
Comment (BradleyBoutcher) [reply CHANGELOG.md:14]: Good wording - committed 👍

File: bin/health-check.rb [hunk 7]
```diff
@@ -44,6 +50,19 @@ def error(message)
   end
 end
 
+if ConjurClient.policy != 'root'
+  begin
+    policy_resource = policy_resource(master_api)
+
+    policy_resource_exists = policy_resource.exists
```
[the remark this answers is not in the document]
Comment (BradleyBoutcher) [reply bin/health-check.rb:57]: comment added 👍

File: bin/health-check.rb (hunk 1 above)
Comment (BradleyBoutcher) [reply bin/health-check.rb:61]: Fixed 👍

File: features/health-check.feature (hunk 2 above)
Comment (BradleyBoutcher) [reply features/health-check.feature:19]: Added 👍

File: features/health-check.feature (hunk 3 above)
Comment (BradleyBoutcher) [reply features/health-check.feature:18]: Not needed - removed! 👍

File: features/health-check.feature (hunk 5 above)
Comment (BradleyBoutcher) [reply features/health-check.feature:26]: Not needed - removed! 👍

File: features/health-check.feature (hunk 4 above)
Comment (BradleyBoutcher) [reply features/health-check.feature:23]: Added 👍

**Pull request merged**